### ISSUE TITLE
Delete recordings that are incomplete

### ISSUE THREAD

**Issue by JamesTownend** [jwallet/spy-spotify#276]
Sometimes I change the track on Spotify before a track has finished. I am left with recordings of incomplete tracks and no way to check later which tracks are incomplete. To counter this, it would be good if there was an option to delete incomplete recordings. If it's not possible to detect when a track is skipped, maybe Spytify can compare the recorded track length with the metadata track length and delete the recording if it is significantly less.

**Comment by jwallet** [comment]
i know some people sometimes records part of a song for making a ringtone or a sample, going forward with that request spytify will have to ask everytime if the user wants to delete it, and metadata are not always retrieved or accurate.

**Comment by jwallet** [comment]
maybe more something like
- when recording is done, move detected incomplete files to a different folder.

**Comment by JamesTownend** [comment]
I hadn't thought about people wanting to only record a portion of a song. Yes, moving incomplete songs to a separate folder (or maybe including "incomplete" in the file name) would be a welcome addition!

**Comment by jwallet** [comment]
did you try to increase minimal length, doing so, it will at least delete tracks that are too short, like for instance at least 1m30?

**Comment by JamesTownend** [comment]
I thought about doing that, but some albums have short intro or transition tracks, for example. I suppose it's not the end of the world if those are lost.

**Comment by jwallet** [comment]
are you skipping songs because you listen to it at the same time ?

**Comment by JamesTownend** [comment]
Yeah... Is that not the recommended way of using Spytify? I have Spotify playing most of the day while I work.

**Comment by jwallet** [comment]
the best way to record is to install Virtual Audio Device that comes with Spytify, set both apps with this device (info available in [FAQ](https://jwallet.github.io/spy-spotify/faq.html#isolate-spotify-audio-endpoint) so it won't record system sound like notifications or youtube playing in the background, this way spotify and spytify are isolated. you can use the web player for your daily use https://open.spotify.com/.
My way is to record in a virtual machine or my local server, and before recording I had a playlist already set with the tracks that I want to record, so the month before or so I added tracks to a playlist call "Need to record"

**Comment by JamesTownend** [comment]
Got it, that solves my problems then. Thanks, you can close this feature request if you want to.

**Comment by jwallet** [comment]
some other users, record at night a playlist or an album using their main audio device, so less chance to record random system sounds since spytify mutes everything when starting the session (but does not do it after it started). The next morning you have the re

**Comment by jwallet** [comment]
no need to close, I may still do it

**Comment by JamesTownend** [comment]
Could be useful in case, e.g. a user's internet connectivity is unstable.

**Comment by jwallet** [comment]
don't forget to use Spotify API for better and accurate results see the FAQ

**Comment by philblue1** [comment]
I had a length issue on some of the tracks while recording an audiobook due to unstable whatever (maybe connection or cpu load) for my not-spotify-capable watch. Besides that i like to record while listening and pick out the songs i liked later which i often start over midway when discovering.

So i would like an option like
[x] plausibility check against spotify-api-metadata-song-length
with [seconds] tolerance. (cause you said spotify metadata is not accurate all the time??!)
and [action= delete, move, rename] when failed with decent log info :)
